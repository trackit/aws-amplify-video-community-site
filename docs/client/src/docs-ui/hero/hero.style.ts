import {css} from "emotion";

export const heroStyle = css`
  display: block;
  text-align: center;
  padding: 3rem 1.5rem 1.5rem;
  background-color: var(--bg-color);

  h1 {
    margin-bottom: 0.875rem;
  }

  > p {
    color: var(--font-color-secondary);
    font-weight: 500;
  }
`;
