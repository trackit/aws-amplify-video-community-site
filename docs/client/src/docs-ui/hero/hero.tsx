import {Component, h, Host} from "@stencil/core";
import {heroStyle} from "./hero.style";

@Component({tag: "docs-hero", shadow: false})
export class DocsHero {
  render() {
    return (
      <Host class={heroStyle}>
        <slot name="heading" />
        <slot name="subheading" />
        <slot name="cta" />
      </Host>
    );
  }
}
