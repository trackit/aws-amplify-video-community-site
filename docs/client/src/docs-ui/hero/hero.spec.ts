import {DocsHero} from "./hero";
import {newSpecPage} from "@stencil/core/testing";

describe("docs-hero", () => {
  describe("Render logic", () => {
    it("should render", async () => {
      expect(
        (
          await newSpecPage({
            components: [DocsHero],
            html: `
              <docs-hero>
                <h1 slot="heading">Heading</h1>
                <h3 slot="subheading">Subheading</h3>
                <docs-landing-hero-cta slot="cta" />
              </docs-hero>
            `,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
