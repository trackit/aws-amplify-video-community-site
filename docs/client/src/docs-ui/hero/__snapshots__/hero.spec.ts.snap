// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docs-hero Render logic should render 1`] = `
<docs-hero class="css-13l53i2">
  <!---->
  <h1 slot="heading">
    Heading
  </h1>
  <h3 slot="subheading">
    Subheading
  </h3>
  <docs-landing-hero-cta slot="cta"></docs-landing-hero-cta>
</docs-hero>
`;
